applications/box.md

```diff
@@ -0,0 +1,213 @@
+# Crust Grant Proposal
+
+* **Project Name:** box
+* **Team Name:** Functionland
+* **Payment Address:** 0x42DD4709a336B8cE54b873069A5c09Ed27ec9Dd8
+* **Receivable Token:** USDT
+
+> ⚠️ *The combination of your GitHub account submitting the application and the payment address above will be your unique identifier during the program. Please keep them safe.*
+
+> ⚠️ *When you submit an application for a Grant，You confirm that you have read these Terms and Conditions carefully and that you agree to be bound by them.*
+
+### Overview
+
+Please provide the following:
+
+  * A project name: box
+  
+  * A brief description of your project: 
+  
+  The mainstream developer experience in the world of independent front-end developers is to purchase a managed service provided by a cloud vendor, which abstracts away the complexity of dealing with back-end development and enable front-end developers to focus on end-to-end user experience for their clients. Currently, there is no option for a front-end developer to build on top of IPFS with a "Developer Experience" similar to serverless functions. They need to run a full js-ipfs node on the platform they want to develop on, for example on Android, which has different file system access than iOS or other operating systems. "box" is an IPFS node which aims to separate the server from front-end development and provides front-end APIs for developers to connect to, using Libp2p, giving them the same experience as a cloud server for any "box" node. To give a full application server experience to the developer, two protocols are being developed in "box" for a start:
+
+File Protocol
+
+Data Protocol
+
+Front-end developers work with a "box" like a server without the need to run a full IPFS node on the platform they develop for, and interface with the "box" node to store or fetch data using libp2p. End users, their clients, can treat "box" like a server to keeps their data safe and gives them all functionalities of IPFS like cluster synchronization. If they lose their phone with the application installed, their files are safe in the "box" node.
```

Comment:
```suggestion
Front-end developers work with a "box" like a server without the need to run a full IPFS node on the platform they develop for, and interface with the "box" node to store or fetch data using libp2p. End users, their clients, can treat "box" like a server to keeps their data safe and gives them all functionalities of IPFS like cluster synchronization. If they lose their phone with the application installed, their files are safe in the "box" node.

box serves the following purposes to complement the current ecosystem with a familiar decentralized alternative for the widely popular client-server model:

1- to provide APIs that front-end can communicate with it regardless of platform without running a full ipfs node. This makes app development do-able by front-end developers without needing to understand back-end and/or worrying about running ipfs on the target device.
2- user has a copy of the files locally as well as keeping them on pinning services (we leave this choice to user as we want to enable them run it at zero cost if they choose to)
3- It provides a pre-setup ipfs node for user so that non-tech-savvy users can still benefit from the ipfs
```

Reply:
This is response to: https://github.com/crustio/Crust-Grants-Program/pull/39/files#r669248392

Reply:
Thanks. One more question. How do you deal with user's metadata, such as user file directory? How do you store, maintain, and manage those file directory information?

Reply:
```suggestion
Front-end developers work with a "box" like a server without the need to run a full IPFS node on the platform they develop for, and interface with the "box" node to store or fetch data using libp2p thorugh 'graph' library provided as the box package. We use IPFS default directory support and content addressing for supporting folders and files. For storing meta data and other user data we use DAG-JSON and IPFS together. We have almost finalized file protocol and are still developing the data protocol, which uses a DAG-JSON file stored on IPFS with query, caching and indexing support directly inside graph library. End users, their clients, can treat "box" like a server to keeps their data safe and gives them all functionalities of IPFS like cluster synchronization. If they lose their phone with the application installed, their files are safe in the "box" node. 
```

Reply:
> Thanks. One more question. How do you deal with user's metadata, such as user file directory? How do you store, maintain, and manage those file directory information?

Replied as part of the new commit suggestion I just made: "We use IPFS default directory support and content addressing for supporting folders and files. For storing meta data and other user data we use DAG-JSON and IPFS together. We have almost finalized file protocol and are still developing the data protocol, which uses a DAG-JSON file stored on IPFS with query, caching and indexing support directly inside graph library"